### PR TITLE
feat: Added search services

### PR DESCRIPTION
**Pull request by zagronitay**

### CODE REVIEW

**Review by irmatov — APPROVED** [review]
LGTM

File: resources/services/search/services.go [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+			},
+			{
+				Name:        "provisioning_state",
+				Description: "The state of the last provisioning operation performed on the search service",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("ServiceProperties.ProvisioningState"),
+			},
+			{
+				Name:        "network_rule_set_ip_rules",
+				Description: "A list of IP restriction rules that defines the inbound network(s) with allowing access to the search service endpoint",
```
Comment (amanenk) [inline resources/services/search/services.go:71]: can `inetarray` be used there?

File: resources/services/search/services.go [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+						Resolver:    schema.ParentIdResolver,
+					},
+					{
+						Name:        "private_endpoint_id",
+						Description: "The resource id of the private endpoint resource from Microsoft.Network provider.",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("Properties.PrivateEndpoint.ID"),
+					},
+					{
+						Name:        "private_link_service_connection_state_status",
```
Comment (amanenk) [inline resources/services/search/services.go:145]: a few lines can be shortened
```
  column "private_link_service_connection_state_status" {
    rename = "connection_status"
  }
  column "private_link_service_connection_state_description" {
    rename = "connection_description"
  }
  column "private_link_service_connection_state_actions_required" {
    rename = "connection_actions_required"
  }
```

**Review by amanenk — CHANGES_REQUESTED** [review]

**Review by roneli — COMMENTED** [review]
nothing more to add other than andrii's comments

**Review by roneli — APPROVED** [review]
LGTM

**Review by amanenk — APPROVED** [review]

**Pull request merged**